include/cell.tcc

```diff
@@ -1236,6 +1236,27 @@ void mpm::Cell<Tdim>::compute_nodal_momentum(const Eigen::VectorXd& shapefn,
     nodes_[i]->update_momentum(true, phase, shapefn(i) * pmass * pvelocity);
 }
 
+//! Compute Bbar matrix
+template <unsigned Tdim>
+std::vector<Eigen::MatrixXd> mpm::Cell<Tdim>::compute_bbar(
+    const std::vector<Eigen::MatrixXd>& bmatrix, unsigned phase) {
+
+  // Compute BBar
+  std::vector<Eigen::MatrixXd> bbar_matrix;
+  for (unsigned i = 0; i < this->nnodes(); ++i) {
+    Eigen::Matrix<double, 3 * (Tdim - 1), Tdim> bbari;
+    for (unsigned j = 0; j < Tdim; ++j) {
```

Comment:
The `bbari` has a size of 6x2 for 3D, and 3x2 for 2D but you are only iterating to maximum of Tdim, which isn't correct. Could you please fix that?

Reply:
I think you misread it. The size is 6x3 for 3D `(3 * (Tdim - 1), Tdim)`, and 3x2 for 2D

Reply:
It's not an issue of size, that was just a typo in my comment, the issue is the bbar matrix is not filled correctly

Reply:
I see. I fixed it. Meant to fill it with bmatrix first, then change only the Tdim part because bbar does not interfere with the shear part.